packages/fxa-settings/src/pages/ResetPassword/ResetPasswordConfirmed/container.tsx

```diff
@@ -39,7 +40,13 @@ const ResetPasswordConfirmedContainer = ({
   const [errorMessage, setErrorMessage] = useState('');
 
   const { email, uid, sessionToken, keyFetchToken, unwrapBKey, verified } =
-    sensitiveDataClient.getData('accountResetData');
+    sensitiveDataClient.getDataType(SensitiveData.Key.AccountReset) || {};
+
+  // If we have lost the required bits for OAuth handling, we have to start again.
+  if (!uid || !sessionToken) {
+    navigate('/signin', { replace: true });
```

Comment:
I think you're going to want to use `useNavigateWithQuery` so we keep our query parameters around. I'm also not sure if we want `replace: true` - do we want to remove this page from the browsing history (maybe we do, just wondering)?

Reply:
@LZoog I tried `navigateWithQuery` and it sort of works. When the page refreshes, I see a blank screen but after a second refresh I see the login page that still includes the RP data needed to complete the auth flow.

These are the query params that I extracted during the refresh. Maybe one of these are causing the page to not render? Do you have any hints which could be causing it?

```

showReactApp    = true
deviceId        = f6c8360dc5bc428ca96c1e0e41b12941
flowBeginTime   = 1733265526561
flowId          = 15eb142a03ccd121acfdb8d1ed5a18014a705bb8e87d6a67b9c8e6e550b96184
access_type     = offline
client_id       = dcdb5ae7add825d2
pkce_client_id  = 38a6b9b3a65a1871
redirect_uri    = http://localhost:8080/api/oauth
scope           = profile+openid
action          = signin
state           = b4a683d8ad824e54b644131c237b217b31e3b43e66306fb5467f43f19048d6e3
flow_id         = 15eb142a03ccd121acfdb8d1ed5a18014a705bb8e87d6a67b9c8e6e550b96184
flow_begin_time = 1733265526561
device_id       = f6c8360dc5bc428ca96c1e0e41b12941
email           = tester@gmail.com
```

Reply:
Huh, so after use `currentAccount()` to get the cached uid, sessionToken, email, verified the bug is no longer reproducible.. which makes sense! We are now using the cached uid and sessionToken from local storage so we no longer need to go through the oauth flow again because we have a valid sessionToken.

cc: @vpomerleau